lumicks/pylake/nb_widgets/detail/undo_redo.py

```diff
@@ -0,0 +1,24 @@
+class UndoableState:
```

Comment:
I would suggest a different name... 

Maybe `StateHistory`? I've also seen people calling this type of thing `Memento`.

Reply:
UndoStack

Reply:
I like `UndoStack`